### DISCUSSION TITLE
Pop-up window

### DISCUSSION THREAD

**Discussion by AndrewCrudin** [alefragnani/vscode-bookmarks#536] [Q&A] [answered]
Could you make, please, a pop-up window with label text when your mouse cursor over bookmark icon?

Something like this:
![image](https://user-images.githubusercontent.com/104717422/166142519-f20dbdeb-8f4e-4506-9e03-9f0254fba049.png)

**Comment by alefragnani** [comment, marked as answer]
Hi @AndrewCrudin ,

Unfortunately, VS Code does not provide API for that. There are open issues requesting not only the tooltip but also click/right click, but no ETA.

These issues were opened more than 5 years ago, and based on that, I guess we won’t see this being available to the extensions at all. I’m sorry.

Hope this helps

**Reply by AndrewCrudin** [reply]
Thank you for the prompt reply!  Sounds like MS is a big brake shoe!
Tell me please, one thing more... I am new to VScode. If am I able to customize context menu in VS code by myself?
I do a lot of line\block commenting. I am too lazy to hit *CTRL + /* all the time. 
I want to add *Comment line* and *Uncomment line* like your *Bookmarks* which I highly appreciate.

**Reply by alefragnani** [reply]
Well, it is their product/platform so, they choose the priorization.

About the customization you asked, you can’t, at least not as _end user_. You will need a _fairly simple_ extension for that. This extension does that here 

https://github.com/alefragnani/vscode-bookmarks/blob/097333f63120c1e5b1591bb30c0946fce532089c/package.json#L268

And here

https://github.com/alefragnani/vscode-bookmarks/blob/097333f63120c1e5b1591bb30c0946fce532089c/package.json#L355

**Reply by AndrewCrudin** [reply]
Yes , I already investigated a little bit yours bookmarks's package.json file on my computer.
But I wonder  if is somewhere   a detailed manual for what to do for adding items to VS Code context menu?

**Reply by alefragnani** [reply]
Here https://code.visualstudio.com/api/references/contribution-points and https://code.visualstudio.com/api/references/vscode-api
